### PR TITLE
fix: [M3-7249] - Linodes Landing Glitchy Render Behavior

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Are you sick and tried of the Linodes Landing page flashing at you? If so, this PR is for you!

## Changes  🔄
- Use preferences directly to control the SideNav (no longer uses `PreferenceToggle`)
- Gives the `PreferenceToggle` the correct default state value

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115251059/1f80c220-1566-43d5-9901-f52f7451e82d" /> | <video src="https://github.com/linode/manager/assets/115251059/59a81565-318f-403a-a43f-d482e2597764" /> |

### Prerequisites
- Have at least 1 Linode on your account

### Reproduction steps
- Go to `cloud.linode.com` and go to the Linodes Landing page
- Click the Akamai logo many times
- Observe the Linodes table flashing when you click the Akamai logo

### Verification steps 
- Go to `http://localhost:3000/linodes`
- Click the Akamai logo many times
- Observe the Linodes table **does not** flash at you and the UI stays the same

## As an Author I have considered 🤔

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/PreferenceToggle/PreferenceToggle.tsx [hunk 1]
```diff
@@ -39,18 +38,17 @@ export const PreferenceToggle = <T extends unknown>(props: Props<T>) => {
...
-  const [lastUpdated, setLastUpdated] = React.useState<number>(0);
-
   const {
     data: preferences,
     error: preferencesError,
     refetch: refetchUserPreferences,
   } = usePreferences();
 
+  const [currentlySetPreference, setPreference] = React.useState<T | undefined>(
+    value ?? preferences?.[preferenceKey]
```
Comment (bnussman-akamai) [inline packages/manager/src/components/PreferenceToggle/PreferenceToggle.tsx:48]: This is the main fix. `PreferenceToggle` didn't pull a default value from preferences causing the children to be render-blocked 😳

File: packages/manager/src/features/Linodes/LinodesLanding/LinodesLanding.tsx [hunk 2]
```diff
@@ -188,7 +188,6 @@ class ListLinodes extends React.Component<CombinedProps, State> {
         <DocumentTitleSegment segment="Linodes" />
         <ProductInformationBanner bannerLocation="Linodes" />
         <PreferenceToggle<boolean>
-          localStorageKey="GROUP_LINODES"
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Linodes/LinodesLanding/LinodesLanding.tsx:191]: This prop did nothing

**Review by jaalah-akamai — APPROVED** [review]

**Review by abailly-akamai — APPROVED** [review]
Thanks for fixing! confirmed not getting the glitch anymore ✅

**Comment by bnussman-akamai** [comment]
Need to figure out a solution to the e2e failure.

**Comment by tyler-akamai** [comment]
it seems like im the only one experiencing it, it might be on my end.

**Review by abailly-akamai — COMMENTED** [review]
I am not entirely sure this PR does what it claims. I think I understand its intent but this is a complex refactor and am not seeing any change with the loading flicker on this branch?

<video src="https://github.com/linode/manager/assets/130582365/eaf83a5b-8268-47e6-a857-6f0e885f5cab" />

**Pull request closed without merging**